src/iface/ethernet.rs

```diff
@@ -262,6 +323,70 @@ impl<'b, 'c, DeviceT> Interface<'b, 'c, DeviceT>
         InterfaceInner::check_ethernet_addr(&self.inner.ethernet_addr);
     }
 
+    /// Add an address to a list of subscribed multicast IP addresses
+    ///
+    /// Returns whether address was added successfully, and, if
+    /// possible, whether an initial immediate announcement has been
+    /// sent.
+    // timestamp is unused in absence of feature "proto-ipv4"
+    #[allow(unused)]
+    pub fn join_multicast_group<T: Into<IpAddress>>(&mut self, addr: T, timestamp: Instant) -> Result<()> {
+        match addr.into() {
+            #[cfg(feature = "proto-ipv4")]
+            IpAddress::Ipv4(addr) => {
+                let is_not_new = self.inner.ipv4_mcast_groups.insert(addr, ())
+                    .map_err(|_| Error::Exhausted)?
+                    .is_some();
+                if is_not_new {
+                    return Ok(());
+                }
+
+                if let Some(pkt) = self.inner.igmp_report_packet(IgmpVersion::Version2, addr) {
+                    // Send initial membership report
+                    let tx_token = self.device.transmit().ok_or(Error::Exhausted)?;
+                    self.inner.dispatch(tx_token, timestamp, pkt)?;
+                }
+
+                Ok(())
+            }
+            // Multicast is not yet implemented for other address families
```

Comment:
I think you mean `IGMP` is not implemented in other address families. Multicast is a requirement for IPv6. See [MLDv2].

[MLDv2]: https://tools.ietf.org/html/rfc3810

Reply:
MLD is not yet implemented in smoltcp.

Reply:
Ah, this comment made me think it was saying Multicast doesn't exist for other families. My bad.